src/test/functional/com/MyApp/budgetControl/api/TestcontainersConfiguration.java

```diff
@@ -0,0 +1,19 @@
+package com.MyApp.budgetControl.api;
+
+import org.springframework.boot.test.context.TestConfiguration;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+@TestConfiguration
+class TestContainersConfiguration {
+
+  private static final PostgreSQLContainer postgreSQLContainer = new PostgreSQLContainer<>
```

Comment:
Have You tested that container stops correctly after tests are done?

Reply:
Using Singelton Containers Pattern like described in documentation 
https://testcontainers.com/guides/testcontainers-container-lifecycle/
"Testcontainers assigns a special label when creating and starting the containers and uses the Ryuk Container behind the scenes to remove containers with that label once the JVM process running the tests exited."